src/class/SimpleData.ts

```diff
@@ -83,15 +81,36 @@ export default class SimpleData {
     get keys() {
         return this._keys
     }
-    set keys(data) {
-        this._keys = data[0] === undefined ? [] : Object.keys(data[0])
+    set keys(keys) {
+        this._keys = keys === undefined ? [] : keys
     }
 
     #updateSimpleData(data: SimpleDataItem[]) {
         this._data = data
         this._keys = data[0] === undefined ? [] : Object.keys(data[0])
     }
 
+    @logCall()
+    loadLocalFile({
+        path,
+        missingKeyValues = { "null": null, "NaN": NaN, "undefined": undefined },
+        encoding = "utf8"
```

Comment:
I am not sure what's the usage here. 

I can't think of usecases where you would want to instantiate `SimpleData` with no data or path. Instead, I would suggest modifying the constructor to accept either `data` or `path`.

Something like this:
```
constructor(
        {
            data = [],
            path,
            missingKeyValues = { "null": null, "NaN": NaN, "undefined": undefined },
            encoding = "utf8"
            ...
        }: {
            data?: SimpleDataItem[],
             path: string,
             missingKeyValues?: SimpleDataItem,
             encoding?: BufferEncoding
            ...
        } = {}) {
        if (data === undefined){
              if (path === undefined){
                   throw an error !
              }
              data = loadLocalFile_({path, verbose, missingKeyValues, encoding})
        }
        data.length > 0 && checkKeys(data)

        this._data = data
```

Reply:
I think it's better to keep them separate because it will be easier to deal with the loadData methods for NodeJS and Web separately.